### ISSUE TITLE
[charts] Support animation

### ISSUE THREAD

**Issue by alexfauquette** [mui/mui-x#9359]
Currently, charts are rendered dry. We could add animation on:
- first render
- container resize
- axis resize
- data update

**Comment by alexfauquette** [comment]
## Recharts

They rely on `react-smooth`. See for example the animated bar elements

https://github.com/alexfauquette/recharts/blob/337081b2633ceb228a0f5fc67044d4f45ff60426/src/cartesian/Bar.tsx#L319

## Nivo

The use '@react-spring/web'

- In plot elements ([Bar](https://github.com/plouc/nivo/blob/master/packages/bar/src/Bar.tsx))
- The [axes components](https://github.com/plouc/nivo/tree/master/packages/axes/src/components) realising the axis in the gris might be defined together to have the grid and the axes tick moving together. Recharts don't have this problem since it does not animate axes modification

**Comment by jamiedruce** [comment]
Big plus one for this feature, but understand its bottom of the list in terms of functionally important features.

Coming from Nivo, the transitions when switching data out is very slick, something we make use of often.

**Comment by alexfauquette** [comment]
@jamiedruce Is customization a thing for you or do you always use the default animation?

**Comment by jamiedruce** [comment]
We just use the default actually. For us, with a 'dashboard' type app, it's just about giving a bit of life to the page where there can be lots of information to digest. We've found it makes a real difference to how the page feels for users.
And certainly for changing data for charts where there are multiple views, it certainly helps communicate the change has happened.

Before Nivo, we used chartjs.org, again with some simple animation of initial load and data changes.
